src/main.rs

```diff
@@ -243,42 +369,41 @@ mod tests {
     }
 
     #[test]
-    fn environments_commands_validate_config() {
-        let mut cmd = cmd();
-        cmd.env("CT_API_KEY", "")
-            .args(&["environments", "list"])
-            .assert()
-            .failure()
-            .stderr(starts_with("The API key is missing."));
-    }
-
-    #[test]
-    fn parameters_commands_validate_config() {
-        let mut cmd = cmd();
-        cmd.env("CT_API_KEY", "")
-            .args(&["parameters", "list"])
-            .assert()
-            .failure()
-            .stderr(starts_with("The API key is missing."));
-    }
-
-    #[test]
-    fn templates_commands_validate_config() {
-        let mut cmd = cmd();
-        cmd.env("CT_API_KEY", "")
-            .args(&["templates", "list"])
-            .assert()
-            .failure()
-            .stderr(starts_with("The API key is missing."));
+    fn need_api_key() {
+        let commands = &[
+            vec!["parameters", "list"],
```

Comment:
I guess this is fine. I preferred separate tests because each can break independently of the others. If you're unhappy with the duplication, it'd be a good candidate for a test macro.

Reply:
I will look into test macros another time... IMO, the original had too much duplicated code.